src/safeds/data/tabular/containers/_table.py

```diff
@@ -1059,7 +1059,7 @@ def add_rows(self, rows: list[Row] | Table) -> Table:
         if isinstance(rows, Table):
             if rows.number_of_rows == 0:
```

Comment:
I guess if we check `number_of_columns` in the next condition, we should also check it here:

```suggestion
            if rows.number_of_columns == 0 or rows.number_of_rows == 0:
```